### DISCUSSION TITLE
I will donate 1000$ for a coil winding development.

### DISCUSSION THREAD

**Discussion by kentthoresen** [vlachoudis/bCNC#1689] [Ideas]
I recently have found the need for a coil winder, i have used bCNC before and its very efficient. 
I had a protoner CNC board lying around and i figured i could ask if there is any interest in developing a coil winding function. 

The coild i need to wind are not circular and they are conical as well so its a bit different than the run of the mill winders alreay on the internet.

The winder needs to use a mandrel (stl file or something) for the coil basic shape. And the possibility to add pauses to add insulation, binder, voltage taps etc. 
Also start position and start direction is a requirement

The winder would need to accept input for wire form, Square, Round, Flat. and well as dimensions 3mm diameter, 2x2mm 1,5x3mm etc.

The x axis needs to move to macth the width of the wire of course and the y axis needs to follow the shape of the mandrel (oval, semi square, paralellogram) according to the mandre model. And it would need to account for each layer added in hight. The z axis is the rotational axis of course, 

Of course the machine layout/configuration would need to be different.

Anybody interested?

I will donate once a functional beta version is released. 

I found this firmware that probably also needs to be updated https://github.com/daveansell/grbl-coil-winder

Update: I guess there is no interest in this i will try to solve this some other way

**Comment by MARIOBASZ** [comment]
If you make a plugin with the corresponding fields to enter the necessary parameters, you can make it generate the code you need. I think it is not very complicated.

**Comment by kentthoresen** [comment]
If i knew how to solve this i wouldnt offer money for it.

**Reply by MARIOBASZ** [reply]
Have you been able to advance?
Hope so! However, if the answer is no, I can try to help, send me an email. Hug.

**Reply by kentthoresen** [reply]
Not solved yet, but  i am invesstigating some options around GrblGru. I am travelling, i will send you a an email when i am back

**Reply by Ams-Laser** [reply]
Hello, i have built some coil winders before, i might be able to help you, feel free to contact me cnclasermx (at) gmail dot com
